backendApi/build.gradle

```diff
@@ -1,3 +1,5 @@
 dependencies {
     vavr()
+
+    implementation project(":branchLayoutApi")
```

Comment:
Actually, regardless of whether we leave it this way or not, in the current shape it should be `api`.

Reply:
https://docs.gradle.org/current/userguide/java_library_plugin.html:

![image](https://user-images.githubusercontent.com/3383210/80203125-f2e20780-8626-11ea-8c2a-b85159ac4dd9.png)

`IBranchLayout branchLayout` is a param of `IGitMacheteRepositoryFactory#create`